### PR TITLE
sql: update the description of auto id conflicts

### PR DESCRIPTION
**Pull request by lilin90**
@shenli @CaitinChen PTAL
Via: https://github.com/pingcap/docs-cn/pull/758

### CODE REVIEW

File: sql/mysql-compatibility.md [hunk 1]
```diff
@@ -32,18 +32,22 @@ The auto-increment ID feature in TiDB is only guaranteed to be automatically inc
...
+> Assume that you have a table with the auto-increment ID:
 > 
 > ```
 > create table t(id int unique key auto_increment, c int);
 > ```
+> 
+> The principle of the auto-increment ID in TiDB is that each tidb-server instance caches a section of ID values for allocation and fetches the next section after this section is used up.
 >
-> The statement is executed as follows:
+> Assume that the cluster contains two tidb-server instances, namely Instance A and Instance B. Instance A caches the auto-increment ID of [1,5000], while Instance B caches the auto-increment ID of [5001,10000].
```
Comment (CaitinChen) [inline sql/mysql-compatibility.md]: [1,5000] -> [1, 5000] 
[5001,10000] ->  [5001, 10000]

File: sql/mysql-compatibility.md [hunk 2]
```diff
@@ -32,18 +32,22 @@ The auto-increment ID feature in TiDB is only guaranteed to be automatically inc
...
+> 
+> The operations are executed as follows:
 >
-> 1. The client inserts a statement to TiDB B which sets the `id` to be 1 and the statement is executed successfully.
-> 2. The client inserts a record to TiDB A which sets the `id` set to the default value 1. In this case, it returns `Duplicated Error`.
+> 1. The client inserts the `insert into t values (1, 1)` statement to Instance B which sets the `id` to 1 and the statement is executed successfully.
+> 2. The client inserts the `insert into t (c) (1)` statement to Instance A. This statement does not specify the value of `id`, therefore Instance A allocates the value. Currently, Instances A caches the auto-increment ID of [1,5000], so it allocates the `id` value to 1 and adds 1 to the local counter. However, at this time the data with the `id` of 1 already exists in the cluster, therefore it reports `Duplicated Error`.
```
Comment (CaitinChen) [inline sql/mysql-compatibility.md]: ditto

**Comment by lilin90** [comment]
@CaitinChen Comments addressed. PTAL

File: sql/mysql-compatibility.md [hunk 3]
```diff
@@ -32,18 +32,22 @@ The auto-increment ID feature in TiDB is only guaranteed to be automatically inc
...
+> The principle of the auto-increment ID in TiDB is that each tidb-server instance caches a section of ID values for allocation and fetches the next section after this section is used up.
 >
-> The statement is executed as follows:
+> Assume that the cluster contains two tidb-server instances, namely Instance A and Instance B. Instance A caches the auto-increment ID of [1, 5000], while Instance B caches the auto-increment ID of [5001, 10000].
+> 
+> The operations are executed as follows:
 >
-> 1. The client inserts a statement to TiDB B which sets the `id` to be 1 and the statement is executed successfully.
-> 2. The client inserts a record to TiDB A which sets the `id` set to the default value 1. In this case, it returns `Duplicated Error`.
+> 1. The client inserts the `insert into t values (1, 1)` statement to Instance B which sets the `id` to 1 and the statement is executed successfully.
```
Comment (queenypingcap) [inline sql/mysql-compatibility.md]: The client inserts -> The client issues

Comment (queenypingcap) [inline sql/mysql-compatibility.md]: The client inserts -> The client issues

File: sql/mysql-compatibility.md [hunk 4]
```diff
@@ -32,18 +32,22 @@ The auto-increment ID feature in TiDB is only guaranteed to be automatically inc
...
+> 2. The client inserts the `insert into t (c) (1)` statement to Instance A. This statement does not specify the value of `id`, therefore Instance A allocates the value. Currently, Instances A caches the auto-increment ID of [1, 5000], so it allocates the `id` value to 1 and adds 1 to the local counter. However, at this time the data with the `id` of 1 already exists in the cluster, therefore it reports `Duplicated Error`.
```
Comment (queenypingcap) [inline sql/mysql-compatibility.md]: Ditto

Comment (queenypingcap) [reply sql/mysql-compatibility.md]: therefore -> so

**Comment by lilin90** [comment]
@queenypingcap PTAL again

**Comment by queenypingcap** [comment]
LGTM

**Pull request merged**